shared/tests/claimify/test_integration.py

```diff
@@ -0,0 +1,534 @@
+    SelectionResult,
+    DisambiguationResult,
+    DecompositionResult,
+    NodeType
+)
+
+# Import integration components conditionally
+try:
+    from clarifai_shared.claimify.integration import (
+        ClaimifyGraphIntegration,
+        create_graph_manager_from_config
+    )
+    from clarifai_shared.graph.neo4j_manager import Neo4jGraphManager
+    from clarifai_shared.graph.models import ClaimInput, SentenceInput
+    NEO4J_AVAILABLE = True
+except ImportError:
+    # Handle case where neo4j is not installed
+    NEO4J_AVAILABLE = False
+    ClaimifyGraphIntegration = None
+    create_graph_manager_from_config = None
+    Neo4jGraphManager = None
+    ClaimInput = None
+    SentenceInput = None
+
+
+class MockNeo4jGraphManager:
+    """Mock implementation of Neo4jGraphManager for testing."""
+    
+    def __init__(self):
+        self.created_claims = []
+        self.created_sentences = []
+    
+    def create_claims(self, claim_inputs):
+        """Mock create_claims method."""
+        # Simulate successful creation by storing inputs
+        self.created_claims.extend(claim_inputs)
+        return [f"claim_{i}" for i in range(len(claim_inputs))]
+    
+    def create_sentences(self, sentence_inputs):
+        """Mock create_sentences method."""
+        # Simulate successful creation by storing inputs
+        self.created_sentences.extend(sentence_inputs)
+        return [f"sentence_{i}" for i in range(len(sentence_inputs))]
+    
+    def setup_schema(self):
+        """Mock setup_schema method."""
+        pass
+
+
+class TestClaimifyGraphIntegration(unittest.TestCase):
+    """Test Claimify to Neo4j integration functionality."""
+    
+    def setUp(self):
+        """Set up test data and integration instance."""
+        if not NEO4J_AVAILABLE:
+            self.skipTest("Neo4j dependencies not available")
+            
+        # Mock the neo4j dependency if not available
+        try:
+            from ..graph.neo4j_manager import Neo4jGraphManager
+            # In tests, we'll use a mocked manager
+            self.graph_manager = MockNeo4jGraphManager()
+        except ImportError:
+            # If neo4j is not available, use the mock
+            self.graph_manager = MockNeo4jGraphManager()
+        
+        self.integration = ClaimifyGraphIntegration(self.graph_manager)
+        
+        # Create test sentence chunks
+        self.test_chunk = SentenceChunk(
+            text="The system failed at startup.",
+            source_id="blk_001",
+            chunk_id="chunk_001",
+            sentence_index=0
+        )
+        
+        self.test_context = ClaimifyContext(current_sentence=self.test_chunk)
+    
+    def test_integration_initialization(self):
+        """Test integration initialization."""
+        integration = ClaimifyGraphIntegration(self.graph_manager)
+        self.assertEqual(integration.graph_manager, self.graph_manager)
+    
+    def test_convert_successful_claim_result(self):
```

Comment:
______ TestClaimifyGraphIntegration.test_convert_successful_claim_result _______
shared/tests/claimify/test_integration.py:143: in test_convert_successful_claim_result
    self.assertTrue(claim.verifiable)
                    ^^^^^^^^^^^^^^^^
E   AttributeError: 'ClaimInput' object has no attribute 'verifiable'